### PR TITLE
Change broader term classifier to a classification scheme classifier

### PR DESCRIPTION
**Pull request by jonescc**
which creates category paths to be indexed for facetted searching by looking up the parent categories for terms defined in a vocabulary thesaurus
in a classification scheme thesaurus

### CODE REVIEW

**Comment by jonescc** [comment]
Requires changes to chef to support configuration of vocabulary and classification thesauri to use to index parameter and platform facets and the modified config-summary.xml configuration

File: web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
+//=============================================================================
```
Comment (danfruehauf) [inline web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java]: Is that an imported file?

File: web/src/test/java/org/fao/geonet/test/TestThesaurusManager.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package org.fao.geonet.test;
```
Comment (danfruehauf) [inline web/src/test/java/org/fao/geonet/test/TestThesaurusManager.java]: I think the filename should be `ThesaurusManagerTest.java`, no?

Comment (jonescc) [inline web/src/test/java/org/fao/geonet/test/TestThesaurusManager.java]: Well.  Its not a test - its a thesaurusmanger for use by tests.

Comment (jonescc) [inline web/src/test/java/org/fao/geonet/test/TestThesaurusManager.java]: Perhaps DirectoryThesaurusLoader would be better

File: web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java (hunk 1 above)
Comment (jonescc) [inline web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java]: Not sure what you are referring to here web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java or the Geonetwork copyright notice.

Comment (danfruehauf) [inline web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java]: The file itself, and whether it should have this copyright.

Comment (jonescc) [inline web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java]: Well I wrote the class so perhaps not.

Comment (jonescc) [inline web/src/main/java/org/fao/geonet/kernel/search/classifier/ClassificationScheme.java]: Followed this up with @pblain who wlll check the requirement here.  GeoNetwork have specific requirements re licensing for contributed code.

**Comment by jonescc** [comment]
Removed FAO license and renamed TestThesaurusManager to ThesaurusDirectoryLoader.  Also reverted delete of BroaderTerm classifier - not sure why I deleted it in the first place and it simplifies migration

**Pull request merged**